### PR TITLE
Refactor/default deployment

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [x] PR title descriptive (can be used in release notes)

## Description
This is the first in 2 PRs to remove hardcoded plh_global data from the repo (next #2067). 

**Main Changes**
- Add support in github actions to pull remote repos
- Add support in github actions for encrypted deployment
- Improve github action variable handling to use environment contexts 
- Make default github actions use debug repo

**Additional changes**
- Add a `public.json` file to deployment folder to keep reference to any _public_ deployments (to allow easier discovery, mostly just for internal use for now)

## Dev Notes
Originally this had been one single PR but decided better to split the gh actions changes from app-data removal for cleaner diffs.

All github actions now target the `debug` deployment as a default. This can be seen in the `test - preview` action deployment listed below

## Author Notes
In order to support pulling remote repo and decrypting with private key the main web-build action has been updated to make use of environment contexts (like the appetize action). This adds support for passing `DEPLOYMENT_PRIVATE_KEY` secret and `DEPLOYMENT_REPO` variables in the environment. Currently this is only needed for using the remote `debug` deployment as default for actions

See example in new `debug` environment
https://github.com/IDEMSInternational/parenting-app-ui/settings/environments

It's still expected that most of these environments will eventually move to their own content repos, but still worth having the example case.

## Review Notes
There aren't any frontend code changes, however would be good to monitor actions on next deployment to make sure correct deployment loaded as expected

## Git Issues

Closes #

## Screenshots/Videos
**Github actions now reference environment (deployment) name during actions**
image

**PR Preview now includes debug repo data**
image

### CODE REVIEW

**Review by esmeetewinkel — APPROVED** [review]
Looks sensible. @jfmcquade Can you resolve the merge conflicts (on `.github/workflows/web-build.yml`) when you review this? I'm guessing I just need to accept all on the PR branch, and discard all `master`, but I'll leave it to you just in case.

**Review by jfmcquade — APPROVED** [review]
Nice to have this, the code looks good.

I resolved multiple merge conflicts in [web-build.yml](https://github.com/IDEMSInternational/parenting-app-ui/pull/2066/files#diff-19b023023d9d583fbe12c89870c1353adccb781f831111f536a839655eaecf94) so @chrismclarke maybe just give that file a once-over before merging.

There are also some formatting changes applied to [audio.component.html](https://github.com/IDEMSInternational/parenting-app-ui/pull/2066/commits/e56cfdfe76dad4d8586c578df0a823dda46d7a1d#diff-6421bcc4dc8d6ebf63a8e22e3163333f47659953f84d5e3e3b1bc910dc01076b) that snuck in somehow when I merged master into this feature branch. They found themselves into another PR of mine too, so I'm happy to have them included here if they'll be quiet afterwards.

I've just noticed that the [action run](https://github.com/IDEMSInternational/parenting-app-ui/actions/runs/6263541526) gives the warning "Cache save failed.", not sure if that's an issue
Screenshot 2023-09-21 at 16 15 25

**Comment by chrismclarke** [comment]
> I've just noticed that the [action run](https://github.com/IDEMSInternational/parenting-app-ui/actions/runs/6263541526) gives the warning "Cache save failed.", not sure if that's an issue
@jfmcquade - No this shouldn't be an issue as the failure was due to an identical cache already existing so not an issue that it did not recreate. I'm guessing this may be due to another action running in parallel, in which case still more time-efficient to allow the parallel operations and duplicate cache creation than running one after the other.

**Pull request merged**
